actionpack/lib/abstract_controller/rendering.rb

```diff
@@ -107,31 +106,24 @@ def _process_options(options)
       options
     end
 
-    # Process the rendered format.
-    # :api: private
-    def _process_format(format)
-    end
-
-    def _set_html_content_type # :nodoc:
-    end
-
-    def _set_rendered_content_type(format) # :nodoc:
-    end
+    private
+      def set_html_content_type # :nodoc:
+        self.content_type = Mime[:html].to_s
+      end
 
-    # Normalize args and options.
-    # :api: private
-    def _normalize_render(*args, &block)
-      options = _normalize_args(*args, &block)
-      #TODO: remove defined? when we restore AP <=> AV dependency
-      if defined?(request) && request.variant.present?
```

Comment:
We have to move down this to the stack to Action Controller or Action View. Mailers for example don't know about request neither variants.